spec/services/proceeding_type_full_text_search_spec.rb

```diff
@@ -0,0 +1,141 @@
+        end
+      end
+
+      context 'search for category of law' do
+        let(:search_term) { 'Family' }
+
+        it 'returns everything' do
+          result_set = subject
+          expect(result_set.count).to eq ProceedingType.count
+        end
+      end
+
+      context 'searching for a term that only exists on one record' do
+        let(:search_term) { 'mutilation' }
+
+        it 'returns one result row' do
+          result_set = subject
+          expect(result_set.size).to eq 1
+        end
+
+        it 'returns an instance of Result' do
+          result = subject.first
+          expect(result).to be_an_instance_of(ProceedingTypeFullTextSearch::Result)
+        end
+
+        it 'returns FGM Protection order' do
+          result = subject.first
+          expect(result.meaning).to eq 'FGM Protection Order'
+          expect(result.description).to eq 'To be represented on an application for a Female Genital Mutilation Protection Order under the Female Genital Mutilation Act.'
+        end
+      end
+
+      context 'search term only exists in additional_search_terms' do
+        let(:search_term) { 'cao' }
+
+        it 'returns two records' do
+          result_set = subject
+          expect(result_set.map(&:ccms_code).sort).to eq %w[SE013 SE014]
+        end
+      end
+
+      context 'searching for a term which occurs in more than one proceeding' do
+        let(:search_term) { 'injunction' }
+
+        it 'returns three results' do
+          result_set = subject
+          expect(result_set.size).to eq 3
+        end
+
+        it 'returns the one with the search term in meaning first' do
+          result_set = subject
+          expect(result_set.map(&:meaning)).to eq ['Harassment - injunction', 'Inherent jurisdiction high court injunction', 'Non-molestation order']
+        end
+
+        context 'when you send one of the codes as an excluded_term' do
+          subject { described_class.call(search_term, excluded_terms) }
+
+          let(:excluded_terms) { 'DA001' }
+
+          it 'returns two results' do
+            result_set = subject
+            expect(result_set.size).to eq 2
+          end
+
+          it 'returns the one with the search term in meaning first and excludes the second result' do
+            result_set = subject
+            expect(result_set.map(&:meaning)).to eq ['Harassment - injunction', 'Non-molestation order']
+          end
+        end
+      end
+
+      context 'multiple term searches' do
+        let(:search_term) { 'protection order' }
+
+        it 'only returns results matching both terms' do
+          result_set = subject
+          expect(result_set.map(&:meaning)).to match_array(['FGM Protection Order',
+                                                            'Forced marriage protection order',
+                                                            'Variation or discharge under section 5 protection from harassment act 1997'])
```

Comment:
I may be misunderstanding the test here, it says 'only returns results matching **both** terms', the 3rd result in the array does not include 'order' so I would assume that it would not be returned/included in the results.

Reply:
This is a copy/paste from Apply with no changes... 😬 
I could change the description to `returns results matching either terms`? Which is what is actually happening?